src/v2/lib/map-error-response.js

```diff
@@ -23,6 +23,8 @@ commandMap.set('UniqueConstraintViolation', transformConflict);
  * @return {Error} Boom error
  */
 const mapErrorResponse = error => {
+  error.name = error.code === '23503' ? 'ForeignKeyConstraintViolation' : error.name;
```

Comment:
This doesn't conform to the pattern we have set up, the service layer should be throwing our service-defined errors for detection here.

Reply:
done - thanks for explaining this one.